src/main/kotlin/calculator/ExpressionParser.kt

```diff
@@ -0,0 +1,33 @@
+package calculator
+
+import calculator.domain.Expression
+import calculator.domain.Operand
+
+class ExpressionParser {
+
+    private fun findCustomDelimiter(input: String): MatchResult? {
+        return CUSTOM_DELIMITER.toRegex().find(input)
+    }
+
+    private fun toExpression(tokens: List<String>): Expression {
+        return Expression(tokens.map(::Operand))
+    }
+
+    fun parse(input: String?): Expression {
+        if (input.isNullOrEmpty()) return Expression(listOf(0))
+
+        findCustomDelimiter(input)?.let {
+            val customDelimiter = it.groupValues[1]
+            val tokens = it.groupValues[2].split(customDelimiter)
+            return toExpression(tokens)
+        }
+
+        val tokens = input.split(DELIMITERS.toRegex())
+        return toExpression(tokens)
+    }
```

Comment:
컨벤션을 참고하여 메서드 선언 위치를 수정해보면 좋을것 같습니다~!
[https://kotlinlang.org/docs/coding-conventions.html#class-layout](https://kotlinlang.org/docs/coding-conventions.html#class-layout)

> Do not sort the method declarations alphabetically or by visibility, and do not separate regular methods from extension methods. Instead, put related stuff together, so that someone reading the class from top to bottom can follow the logic of what's happening. Choose an order (either higher-level stuff first, or vice versa) and stick to it.